### PR TITLE
Updating systematics

### PR DESCRIPTION
**Pull request by Ragansu**
This is a PR to include multiple systematics. The settings of the competition can be set in the config.py along with the input dir.

The code also split the get_bootstrap function into 2, one for bootstrapping and the other for only systematics shift script. 

The ranges for each systematics is not set at some random value and needs to be discussed. 

Please give your comments 
Regards
RC

### CODE REVIEW

**Comment by Ragansu** [comment]
The code required Pythia8 process flags for W weights hence that part is work in progress, It needs the new data set to work.

File: ingestion_program/datasets.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+def Neurips2024_public_dataset():
+
+    file = "public_data.zip"
+    if file not in os.listdir(parent_path):
+        subprocess.run(
+            [
+                "wget",
+                "-O",
+                os.path.join(parent_path, "public_data.zip"),
+                "https://codalab.coresearch.club/my/datasets/download/0e2d7e8e-1b8b-4b3f-8b8b-3b3c5d4e4e3d",
```
Comment (ihsaan-ullah) [inline ingestion_program/datasets.py:164]: fix the dataset link

File: ingestion_program/run_ingestion.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
+import sys
```
Comment (ihsaan-ullah) [inline ingestion_program/run_ingestion.py:1]: add examples of usage with different arguments

File: scoring_program/run_scoring.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import sys
```
Comment (ihsaan-ullah) [inline scoring_program/run_scoring.py:1]: add examples with different arguments

**Comment by cjh1** [comment]
Lets not commit the data to the repository. Adding large binary data to a git repo is generally not a good idea

File: ingestion_program/ingestion_parallel.py [hunk 4]
```diff
@@ -286,10 +286,10 @@ def show_duration(self):
         print(f"[✔] Total duration: {self.get_duration()}")
         print("---------------------------------")
 
-    def save_duration(self):
+    def save_duration(self, output_dir=None):
```
Comment (cjh1) [inline ingestion_program/ingestion_parallel.py:289]: This seems to be duplicated between the parallel and sequential version. Common base class?

File: ingestion_program/datasets.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+def Neurips2024_public_dataset():
+    current_path = os.getcwd()
+    file_read_loc = os.path.join(current_path, "public_data")
+    if not os.path.isdir(file_read_loc):
+        os.mkdir(file_read_loc)
+
+    file = "public_data.zip"
+    if file not in os.listdir(file_read_loc):
+        wget.download(
```
[the remark this answers is not in the document]
Comment (Ragansu) [reply ingestion_program/datasets.py:158]: Sounds like a good idea. I just didn't want to use subprocess that's all. Could you implement that change and push it?

[the remark this answers is not in the document]
Comment (cjh1) [reply ingestion_program/datasets.py:158]: Sure

[the remark this answers is not in the document]
Comment (cjh1) [reply ingestion_program/datasets.py]: Done

File: ingestion_program/ingestion_parallel.py (hunk 4 above)
Comment (Ragansu) [reply ingestion_program/ingestion_parallel.py:289]: Yeah ingestion parallel has many things similar to ingestion serial. But I am not personally a fan of inheritance. But if that makes cleaner sure. What do you think @ihsaan-ullah ?

**Review by Ragansu — COMMENTED** [review]
The load data is passed into the model class. the idea was to give the freedom of not loading the training data to save memory and time. But I am not sure I did it correctly.

File: ingestion_program/ingestion_parallel.py (hunk 4 above)
Comment (ihsaan-ullah) [reply ingestion_program/ingestion_parallel.py:289]: Lets keep them sperate for now. Changing something may break something else. In the next update we can clean and remove repetition

**Comment by ihsaan-ullah** [comment]
> Lets not commit the data to the repository. Adding large binary data to a git repo is generally not a good idea

Are you referring to the input data and reference data? 
We need them for starting kit

**Review by ihsaan-ullah — COMMENTED** [review]
Apart from physics stuff, looks good

**Comment by Ragansu** [comment]
I am planning to merge this PR as it has a lot of things other that systematics. This makes it easier to test.  
There were few more bugs in the code like for systematics which I have updated today. 

Any issues detected can be taken care of in another issues


RC

**Comment by Ragansu** [comment]
Merging changes since I tested the model. 
Further testing and improvement will be done by Yulei

**Pull request merged**